File: fix_student.js
const fs = require('fs');
const path = r'c:\Users\harih\OneDrive\Desktop\secure_Internship_management_portal-main\student.html';
let content = fs.readFileSync(path, 'utf8');
let lines = content.split('\n');

let startLine = -1;
for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('if (id) {') && i < lines.length - 10) {
        if (!lines[i + 1].includes('return `https://drive.google.com/uc?export=view&id=${id}`;')) {
            startLine = i;
            break;
        }
    }
}

if (startLine !== -1) {
    console.log(`Found broken area at line ${startLine + 1}`);
    const newContent = [
        '            if (id) {',
        '                return `https://drive.google.com/uc?export=view&id=${id}`;',
        '            }',
        '            return url;',
        '        }',
        '',
        '        function closePromoModal() {',
        "            document.getElementById('promoModal').style.display = 'none';",
        '        }',
        '',
        '        function checkPromotion() {',
        '            const settings = DB.getSettings();',
        '            console.log("Checking Promotion:", settings);',
        '',
        '            // Allow string \'true\' or boolean true',
        "            const isActive = String(settings.promoActive).toLowerCase() === 'true';",
        '',
        '            if (isActive && settings.promoImage) {',
        "                if (!document.getElementById('promoModal')) return;",
        '',
        '                document.getElementById(\'promoImage\').src = getDriveDirectLink(settings.promoImage);',
        '',
        "                const promoLinkBtn = document.getElementById('promoLink');",
        "                const linkUrl = settings.promoLink ? settings.promoLink.trim() : '';",
        '',
        "                if (linkUrl && linkUrl !== '#' && linkUrl !== '') {",
        '                    promoLinkBtn.href = linkUrl;',
        "                    promoLinkBtn.style.display = 'flex'; // Show button",
        '',
        "                    const btnText = settings.promoBtnText || 'Learn More';",
        "                    document.getElementById('promoBtn').innerText = btnText;",
        '                } else {',
        "                    promoLinkBtn.style.display = 'none'; // Hide button if no link",
        '                }',
        '',
        "                document.getElementById('promoModal').style.display = 'flex';",
        '            }',
        '        }',
        '',
        '        (async function initApp() {',
        '',
        '            await DB.initialize();',
        '',
        '            // Re-fetch user in case updates happened since login',
        '            const freshUser = DB.getCurrentUser();',
        '            if (freshUser) {',
        '                // Update local student object reference',
        '                Object.assign(student, freshUser);',
        "                document.getElementById('userName').innerText = student.name;",
        "                document.getElementById('userDomain').innerText = student.domain;",
        '            }',
        '',
        '            renderTasks();',
        '            renderPaymentStatus();',
        '            checkPaymentRequirement();',
        '            // Check for promotions last so it overlays others if needed',
        '            checkPromotion();',
        '',
        '            // Handle Deep Linking (e.g. #refer)',
        "            const hash = window.location.hash.replace('#', '');",
        "            if (hash && ['tasks', 'payment', 'refer', 'help'].includes(hash)) {",
        '                switchTab(hash);',
        '            }',
        '',
        '            // --- Payment Reminder Flow ---',
        "            // If student is approved for offer but hasn't paid, show the gift popup",
        '            if (freshUser) {',
        '              const allTasks = DB.getInternTasks(freshUser.id, freshUser.domain);',
        '              const coreTasksActive = allTasks.filter(t => !t.isCertificateTask);',
        "              const isFinished = coreTasksActive.length > 0 && coreTasksActive.every(t => t.status === 'approved');",
        '',
        '              if (isFinished && (freshUser.offerApproved || freshUser.offerLetterUrl)) {',
        "                  if (freshUser.paymentStatus !== 'verified' && freshUser.paymentStatus !== 'pending') {",
        '                      // Slight delay for better UX',
        '                      setTimeout(() => {',
        "                          document.getElementById('paymentReminderModal').style.display = 'flex';",
        '                      }, 1000);',
        '                  }',
        '              }',
        '            }',
        '',
        '        })();',
        '',
        '        function checkPaymentRequirement() {',
        '            // Notifications disabled based on requirements',
        '        }',
        '',
        '        function closePaymentReminder() {',
        "            document.getElementById('paymentReminderModal').style.display = 'none';",
        '        }',
        '',
        '        async function requestWithdrawal() {',
        "            const btn = document.getElementById('btnWithdraw');",
        '            const originalText = btn.innerText;',
        "            btn.innerText = 'Sending...';",
        '            btn.disabled = true;',
        '',
        '            const currentUser = DB.getCurrentUser();',
        '            // Call DB to update status',
        '            // We use updateInternProfile for this status flag',
        "            const success = await DB.updateInternProfile(currentUser.id, { withdrawalStatus: 'requested', withdrawalDate: new Date().toISOString() });",
        '',
        '            if (success) {',
        "                alert('Withdrawal request sent successfully! You will be notified once processed.');",
        '                renderMyReferrals(); // Re-render to update UI state',
        '            } else {',
        "                alert('Failed to send request. Please try again.');",
        '                btn.innerText = originalText;',
        '                btn.disabled = false;',
        '            }',
        '        }',
        '',
        '        function goToPayment() {',
        '            closePaymentReminder();',
        '            openPaymentModal();',
        '        }',
        '',
        '        function openPaymentModal() {',
        "            document.getElementById('paymentModal').style.display = 'flex';",
        '            renderPaymentStatus();',
        '        }'
    ];

    let endLine = -1;
    for (let i = startLine; i < lines.length; i++) {
        if (lines[i].includes('function closePaymentModal()')) {
            endLine = i;
            break;
        }
    }

    if (endLine !== -1) {
        lines.splice(startLine, endLine - startLine, ...newContent);
        fs.writeFileSync(path, lines.join('\n'), 'utf8');
        console.log('Successfully fixed student.html');
    } else {
        console.log('Could not find end of broken area');
    }
} else {
    console.log('Could not find start of broken area');
}

File: fix_student.py
import sys

with open(r'c:\Users\harih\OneDrive\Desktop\secure_Internship_management_portal-main\student.html', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Reconstruct the broken area
start_line = -1
for i, line in enumerate(lines):
    if 'if (id) {' in line and i < len(lines)-10:
        if 'return `https://drive.google.com/uc?export=view&id=${id}`;' not in lines[i+1]:
             start_line = i
             break

if start_line != -1:
    print(f"Found broken area at line {start_line+1}")
    new_content = [
        '            if (id) {\n',
        '                return `https://drive.google.com/uc?export=view&id=${id}`;\n',
        '            }\n',
        '            return url;\n',
        '        }\n',
        '\n',
        '        function closePromoModal() {\n',
        "            document.getElementById('promoModal').style.display = 'none';\n",
        '        }\n',
        '\n',
        '        function checkPromotion() {\n',
        '            const settings = DB.getSettings();\n',
        '            console.log("Checking Promotion:", settings);\n',
        '\n',
        '            // Allow string ' + "'true' or boolean true\n",
        "            const isActive = String(settings.promoActive).toLowerCase() === 'true';\n",
        '\n',
        '            if (isActive && settings.promoImage) {\n',
        "                if (!document.getElementById('promoModal')) return;\n",
        '\n',
        '                document.getElementById(' + "'promoImage').src = getDriveDirectLink(settings.promoImage);\n",
        '\n',
        "                const promoLinkBtn = document.getElementById('promoLink');\n",
        "                const linkUrl = settings.promoLink ? settings.promoLink.trim() : '';\n",
        '\n',
        "                if (linkUrl && linkUrl !== '#' && linkUrl !== '') {\n",
        '                    promoLinkBtn.href = linkUrl;\n',
        "                    promoLinkBtn.style.display = 'flex'; // Show button\n",
        '\n',
        "                    const btnText = settings.promoBtnText || 'Learn More';\n",
        "                    document.getElementById('promoBtn').innerText = btnText;\n",
        '                } else {\n',
        "                    promoLinkBtn.style.display = 'none'; // Hide button if no link\n",
        '                }\n',
        '\n',
        "                document.getElementById('promoModal').style.display = 'flex';\n",
        '            }\n',
        '        }\n',
        '\n',
        '        (async function initApp() {\n',
        '\n',
        '            await DB.initialize();\n',
        '\n',
        '            // Re-fetch user in case updates happened since login\n',
        '            const freshUser = DB.getCurrentUser();\n',
        '            if (freshUser) {\n',
        '                // Update local student object reference\n',
        '                Object.assign(student, freshUser);\n',
        "                document.getElementById('userName').innerText = student.name;\n",
        "                document.getElementById('userDomain').innerText = student.domain;\n",
        '            }\n',
        '\n',
        '            renderTasks();\n',
        '            renderPaymentStatus();\n',
        '            checkPaymentRequirement();\n',
        '            // Check for promotions last so it overlays others if needed\n',
        '            checkPromotion();\n',
        '\n',
        '            // Handle Deep Linking (e.g. #refer)\n',
        "            const hash = window.location.hash.replace('#', '');\n",
        "            if (hash && ['tasks', 'payment', 'refer', 'help'].includes(hash)) {\n",
        '                switchTab(hash);\n',
        '            }\n',
        '\n',
        '            // --- Payment Reminder Flow ---\n',
        "            // If student is approved for offer but hasn't paid, show the gift popup\n",
        '            if (freshUser) {\n',
        '              const allTasks = DB.getInternTasks(freshUser.id, freshUser.domain);\n',
        '              const coreTasksActive = allTasks.filter(t => !t.isCertificateTask);\n',
        "              const isFinished = coreTasksActive.length > 0 && coreTasksActive.every(t => t.status === 'approved');\n",
        '\n',
        '              if (isFinished && (freshUser.offerApproved || freshUser.offerLetterUrl)) {\n',
        "                  if (freshUser.paymentStatus !== 'verified' && freshUser.paymentStatus !== 'pending') {\n",
        '                      // Slight delay for better UX\n',
        '                      setTimeout(() => {\n',
        "                          document.getElementById('paymentReminderModal').style.display = 'flex';\n",
        '                      }, 1000);\n',
        '                  }\n',
        '              }\n',
        '            }\n',
        '\n',
        '        })();\n',
        '\n',
        '        function checkPaymentRequirement() {\n',
        '            // Notifications disabled based on requirements\n',
        '        }\n',
        '\n',
        '        function closePaymentReminder() {\n',
        "            document.getElementById('paymentReminderModal').style.display = 'none';\n",
        '        }\n',
        '\n',
        '        async function requestWithdrawal() {\n',
        "            const btn = document.getElementById('btnWithdraw');\n",
        '            const originalText = btn.innerText;\n',
        "            btn.innerText = 'Sending...';\n",
        '            btn.disabled = True;\n',
        '\n',
        '            const currentUser = DB.getCurrentUser();\n',
        '            // Call DB to update status\n',
        '            // We use updateInternProfile for this status flag\n',
        "            const success = await DB.updateInternProfile(currentUser.id, { withdrawalStatus: 'requested', withdrawalDate: new Date().toISOString() });\n",
        '\n',
        '            if (success) {\n',
        "                alert('Withdrawal request sent successfully! You will be notified once processed.');\n",
        '                renderMyReferrals(); // Re-render to update UI state\n',
        '            } else {\n',
        "                alert('Failed to send request. Please try again.');\n",
        '                btn.innerText = originalText;\n',
        '                btn.disabled = False;\n',
        '            }\n',
        '        }\n',
        '\n',
        '        function goToPayment() {\n',
        '            closePaymentReminder();\n',
        '            openPaymentModal();\n',
        '        }\n',
        '\n',
        '        function openPaymentModal() {\n',
        "            document.getElementById('paymentModal').style.display = 'flex';\n",
        '            renderPaymentStatus();\n',
        '        }\n'
    ]
    
    # Find where to stop replacing (search for the next stable function like closePaymentModal)
    end_line = -1
    for i in range(start_line, len(lines)):
        if 'function closePaymentModal()' in lines[i]:
            end_line = i
            break
            
    if end_line != -1:
        lines[start_line:end_line] = new_content
        with open(r'c:\Users\harih\OneDrive\Desktop\secure_Internship_management_portal-main\student.html', 'w', encoding='utf-8') as f:
            f.writelines(lines)
        print("Successfully fixed student.html")
    else:
        print("Could not find end of broken area")
else:
    print("Could not find start of broken area")
